src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -96,5 +103,16 @@ internal static ProjectId ReadFrom(ObjectReader reader)
 
             return CreateFromSerialized(guid, debugName);
         }
+
+        internal Checksum Checksum
+            => _lazyChecksum.Initialize(static @this => Checksum.Create(@this,
+                static (@this, writer) =>
+                {
+                    // Combine "ProjectId" string and guid.  That way in the off chance that something in the stack uses
+                    // the same Guid for something like a Solution or Document, that we'll still consider the checksums
+                    // different.
+                    writer.WriteString(nameof(ProjectId));
```

Comment:
I'd be quite surprised if that was the case...

Reply:
sure.  but i don't mind being extremely careful here.  Since guid comes in from external sources, i want to be untrusting of it.